app/models/lti_configuration.rb

```diff
@@ -7,4 +7,19 @@ def self.find_by_auth_hash(hash)
     consumer_key = hash.credentials.token
     find_by(consumer_key: consumer_key)
   end
+
+  def context_membership_url(use_cache: true, nonce: nil)
```

Comment:
I took heavy inspiration for this from #1964, in that it will use the persisted attribute if it exists or fallback on the LTI launch information if it doesn't and update the model accordingly (returning `nil` in the sad path that it exists on neither postgres or the lti launch context).